File: C#/Assignments/Fundamentals/WizardNinjaSamurai/Models/Human.cs
using System;

namespace Human
{
    public class Human
        {
            public string Name;
            public int Strength;
            public int Intelligence;
            public int Dexterity;
            public int Health;
            public int health
            {
                get { return Health; }
            }
            public Human()
            {
                Name = "John";
                Strength = 3;
                Intelligence = 3;
                Dexterity = 3;
                Health = 100;
            }
            public Human(string name, int str, int intel, int dex, int hp)
            {
                Name = name;
                Strength = str;
                Intelligence = intel;
                Dexterity = dex;
                Health = hp;
            }
            public virtual int Attack(Human target)
            {
                int damage = Strength * 5;
                target.Health = target.Health - damage;
                Console.WriteLine($"{Name} hit {target.Name} for {damage}!");
                Console.WriteLine($"{target.Name} has {target.Health} remaining health");
                return Health;
            }
        }
}

File: Ruby/Rails/OOP/WizardNinjaSamurai/human.rb
class Human
    attr_accessor :strength, :intelligence, :stealth, :health, :name
    def initialize(name)
        @name = name
        @strength = 3
        @intelligence = 3
        @stealth = 3
        @health = 100
    end

    def attack(target)
        if target.class.ancestors.include?(Human)
            target.health -= 5
            puts "#{target.name} was attacked and lost 5 health, #{target.health} remaining!"
            true
        else
            puts "Target not attackable!"
            false
        end
    end
end